bazel/envoy_build_system.bzl

```diff
@@ -455,15 +455,17 @@ def envoy_cc_test(
     test_lib_tags = []
     if coverage:
         test_lib_tags.append("coverage_test_lib")
-    envoy_cc_test_library(
-        name = name + "_lib",
+    _envoy_cc_test_infrastructure_library(
+        name = name + "_lib_INTERNAL_ONLY",
         srcs = srcs,
         data = data,
         external_deps = external_deps,
-        deps = deps,
+        deps = deps + [repository + "//test/test_common:printers_includes"],
```

Comment:
Why this change? What do we gain by replacing envoy_cc_test_library?

Reply:
We cannot continue to use `envoy_cc_test_library` here since it declares all targets created with it as `//visibility:public`, which means we cannot restrict the visibility of the `_lib_internal_only` targets to `@envoy//test/coverage:__pkg__`.